### PR TITLE
understanding JS call apply and bind and how they are different

### PR DESCRIPTION
**Pull request by sagarmunjal**
## Call apply and bind
[@munjalsagar](https://twitter.com/munjalsagar)

### CODE REVIEW

**Comment by zenopopovici** [comment]
There are a lot of language issues ... we can't merge it like this.

File: _posts/en/javascript/2017-03-29-understanding-apply-call-bind.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
+---
+layout: post
+
+title: Call apply and bind. and how they are different.
```
Comment (zenopopovici) [inline _posts/en/javascript/2017-03-29-understanding-apply-call-bind.md:4]: Title is weird, maybe "JS Apply, Call, and Bind Methods"

**Comment by sagarmunjal** [comment]
What language issues? Could u elaborate?

File: _posts/en/javascript/2017-03-29-understanding-apply-call-bind.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+categories:
+    - en
+    - javascript
+---
+
+
+
+
+You have three automobiles `your_scooter , your_car and your_jet` which start with the same mechanism (method).
```
Comment (zenopopovici) [inline _posts/en/javascript/2017-03-29-understanding-apply-call-bind.md]: automobiles => vehicles

File: _posts/en/javascript/2017-03-29-understanding-apply-call-bind.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+
+
+
+You have three automobiles `your_scooter , your_car and your_jet` which start with the same mechanism (method). 
+We created an object `automobile` with a method `push_button_engineStart`. 
+
+    var your_scooter, your_car, your_jet;
+    var automobile = {
+    	    push_button_engineStart: function (runtime){
```
Comment (zenopopovici) [inline _posts/en/javascript/2017-03-29-understanding-apply-call-bind.md]: Either camelcase or _

File: _posts/en/javascript/2017-03-29-understanding-apply-call-bind.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    		console.log(this.name + "'s" + ' engine_started, buckle up for the ride for ' + runtime + " minutes");
+    	}
+    }
+
+Lets understand when is call and apply used. Lets suppose that you are an engineer and you have `your_scooter`, `your_car` and `your_jet` which did not come with a push_button_engine_start and you wish to use a third party `push_button_engineStart`.
+
+If you run the following lines of code, they will give an error. WHY?
```
Comment (zenopopovici) [inline _posts/en/javascript/2017-03-29-understanding-apply-call-bind.md]: give => throw

File: _posts/en/javascript/2017-03-29-understanding-apply-call-bind.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+We created an object `automobile` with a method `push_button_engineStart`. 
+
+    var your_scooter, your_car, your_jet;
+    var automobile = {
+    	    push_button_engineStart: function (runtime){
+    		console.log(this.name + "'s" + ' engine_started, buckle up for the ride for ' + runtime + " minutes");
+    	}
+    }
+
+Lets understand when is call and apply used. Lets suppose that you are an engineer and you have `your_scooter`, `your_car` and `your_jet` which did not come with a push_button_engine_start and you wish to use a third party `push_button_engineStart`.
```
Comment (zenopopovici) [inline _posts/en/javascript/2017-03-29-understanding-apply-call-bind.md]: Let's

File: _posts/en/javascript/2017-03-29-understanding-apply-call-bind.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+
+If you run the following lines of code, they will give an error. WHY?
+
+    //your_scooter.push_button_engineStart();
```
Comment (zenopopovici) [inline _posts/en/javascript/2017-03-29-understanding-apply-call-bind.md]: Space after comment

File: _posts/en/javascript/2017-03-29-understanding-apply-call-bind.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+    automobile.push_button_engineStart.call(your_jet,10);
+    automobile.push_button_engineStart.call(your_car,40);
+
+So the above example is successfully gives your_scooter, your_car, your_jet a feature from automobile object.
+ 
+**Let's dive deeper**
+Here we will split the above line of code. 
+`automobile.push_button_engineStart` is helping us to get the method being used. 
+
+Further we use apply or call using the dot notation.
```
Comment (zenopopovici) [inline _posts/en/javascript/2017-03-29-understanding-apply-call-bind.md]: Further doesn't apply here.

File: _posts/en/javascript/2017-03-29-understanding-apply-call-bind.md [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+ 2. arguments
+
+So here we set the context in the final line of code. 
+
+`automobile.push_button_engineStart.apply(your_scooter,[20])`
+
+**Difference between call and apply** is just that apply accepts parameters in the form of an array while call simply can accept a comma separated list of arguments. 
+
+
+**what is JS Bind function?**
```
Comment (zenopopovici) [inline _posts/en/javascript/2017-03-29-understanding-apply-call-bind.md]: Start sentence with uppercase.

**Review by zenopopovici — CHANGES_REQUESTED** [review]

**Comment by zenopopovici** [comment]
There are a lot more. I can't re-write the tip for you. Please get someone to spellcheck it.

**Comment by sagarmunjal** [comment]
I'll make note of the spell check and language and rectify.

Is there anything amiss in the logic explained how apply call and bind
work?

**Comment by zenopopovici** [comment]
That's for the other members to weigh in. I personally think in can be improved. See: https://alexperry.io/personal/2016/04/03/How-to-use-apply-bind-and-call.html

**Review by zenopopovici — APPROVED** [review]

**Review by zenopopovici — COMMENTED** [review]
The title is still not correct.

**Comment by zenopopovici** [comment]
Quality issues

**Pull request closed without merging**